scala/lambdas/ingest-mapper/src/main/scala/uk/gov/nationalarchives/ingestmapper/MetadataService.scala

```diff
@@ -88,7 +88,26 @@ class MetadataService(s3: DAS3Client[IO], discoveryService: DiscoveryService) {
           Stream.evals {
             topLevelIdsToDepartmentSeries.map { itemToDepartmentSeries =>
               val parentPaths = getParentPaths(json, itemToDepartmentSeries)
-              val updatedJson = jsonArr.map { metadataEntry =>
+              val childCountMap = parentPaths.groupBy(_._2).view.mapValues(_.size).toMap
+
+              def addChildCount(obj: Obj) = {
+                val values = obj.value.toMap
```

Comment:
Maybe `entry`, `item` or `attributes` rather than `values`?

Reply:
`attributes` is ok.